src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart();
```

Comment:
I think it's probably easier to use if the stage is passed in (in case you want to subscribe to multiple stages)... Although that would require the interface to be generic. What do you say?